update/v1alpha1/types_machineconfigpool_progress_insight.go

```diff
@@ -0,0 +1,173 @@
+	// +optional
+	Status MachineConfigPoolProgressInsightStatus `json:"status"`
+}
+
+// MachineConfigPoolProgressInsightSpec is empty for now, MachineConfigPoolProgressInsightSpec is purely status-reporting API. In the future spec may be used
+// to hold configuration to drive what information is surfaced and how
+type MachineConfigPoolProgressInsightSpec struct {
+}
+
+// MachineConfigPoolProgressInsightStatus reports the state of a MachineConfigPool resource (which represents a pool of nodes
+// update in standalone clusters), during a cluster update.
+// +kubebuilder:validation:XValidation:rule="self.name == self.resource.name",message=".name must match .resource.name"
+type MachineConfigPoolProgressInsightStatus struct {
+	// conditions provide details about the machine config pool update. It contains at most 10 items. Known conditions are:
+	// - Updating: whether the pool is updating; When Updating=False, the reason field can be Pending, Updated or Excluded
+	// +listType=map
+	// +listMapKey=type
+	// +patchStrategy=merge
+	// +patchMergeKey=type
+	// +optional
+	// +kubebuilder:validation:MaxItems=10
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// name is the name of the machine config pool
+	// +required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=64
+	// +kubebuilder:validation:Pattern=`^[a-z0-9-]+$`
+	Name string `json:"name"`
+
+	// resource is the MachineConfigPool resource that represents the pool
+	//
+	// +Note: By OpenShift API conventions, in isolation this should be a specialized reference that refers just to
+	// +resource name (because the rest is implied by status insight type). However, because we use resource references in
+	// +many places and this API is intended to be consumed by clients, not produced, consistency seems to be more valuable
+	// +than type safety for producers.
+	// +required
+	// +kubebuilder:validation:XValidation:rule="self.group == 'machineconfiguration.openshift.io' && self.resource == 'machineconfigpools'",message="resource must be a machineconfigpools.machineconfiguration.openshift.io resource"
+	Resource ResourceRef `json:"resource"`
+
+	// scopeType describes whether the pool is a control plane or a worker pool
+	// +required
+	Scope ScopeType `json:"scopeType"`
+
+	// assessment is the assessment of the machine config pool update process. Valid values are: Pending, Completed, Degraded, Excluded, Progressing
+	// +required
+	Assessment PoolAssessment `json:"assessment"`
+
+	// completion is a percentage of the update completion (0-100)
+	// +required
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=100
+	Completion int32 `json:"completion"`
+
+	// summaries is a list of counts of nodes matching certain criteria (e.g. updated, degraded, etc.). Maximum 16 items can be listed.
```

Comment:
Since the type is an enum, the maximum should be tied to the number of different possible enum values